.github/workflows/tests.yml

```diff
@@ -0,0 +1,106 @@
+name: CI-Tests
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - 'macos-10.15'
+          - 'macos-11.0'
+          - 'ubuntu-18.04'
+          - 'ubuntu-20.04'
+          - 'ubuntu-16.04'
+        python-version:
+          - '3.7'
+          - '3.8'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set XCode version
+        if: contains(matrix.os, 'macos-11.0')
+        run: sudo xcode-select -s "/Applications/Xcode_12.2.app"
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Package installation
+        run: |
+          echo "shimming-toolbox installation"
+          pip install . || exit "$?"
+
+      - name: dcm2niix macOS
+        if: contains(matrix.os, 'macos')
+        run: brew install dcm2niix
+
+      - name: dcm2niix Ubuntu
+        if: contains(matrix.os, 'ubuntu')
+        run: |
+          curl -JLO https://github.com/rordenlab/dcm2niix/releases/latest/download/dcm2niix_lnx.zip || exit "$?"
+          unzip -o dcm2niix_lnx.zip || exit "$?"
+          sudo install dcm2nii* /usr/bin/ || exit "$?"
```

Comment:
You tagged me to review the last version but my only comment on that would've been that you'd downloaded the linux app even on macOS, but you've got that covered here :)